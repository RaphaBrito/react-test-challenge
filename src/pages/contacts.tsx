import { AppError } from "../components/app-error";
import { ContactCard } from "../components/contact-card";
import { CreateContactCard } from "../components/create-contact-card";
import { Loading } from "../components/loading";
import {
  useContacts,
  useContactsCreateMutation,
  useContactsDeleteMutation,
  useContactsEditMutation,
} from "../hooks/contacts";
import styles from "../styles/list.module.css";

import type { Contact, ContactFormData } from "../types/contact";

export function Contacts() {
  const { contacts = [], isPending, isError } = useContacts();
  const createMutation = useContactsCreateMutation();
  const editMutation = useContactsEditMutation();
  const deleteMutation = useContactsDeleteMutation();

  const handleCreate = (formData: ContactFormData) => {
    createMutation.mutate(formData);
  };

  const handleEdit = (contact: Contact) => {
    editMutation.mutate(contact);
  };

  const handleDelete = (id: number) => {
    deleteMutation.mutate(id);
  };

  if (isPending) {
    return <Loading />;
  }

  if (isError) {
    return <AppError />;
  }

  return (
    <div>
      <h1>Contatos</h1>
      <div className={styles.listGrid}>
        {contacts.map((contact: Contact) => (
          <ContactCard
            key={contact.id}
            contact={contact}
            onDelete={handleDelete}
            onEdit={handleEdit}
          />
        ))}

        <CreateContactCard onCreate={handleCreate} />
      </div>
    </div>
  );
}
