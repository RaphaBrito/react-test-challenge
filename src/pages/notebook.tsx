import { AppError } from "@/components/app-error";
import { CreateNoteCard } from "@/components/create-note-card";
import { Loading } from "@/components/loading";
import { NoteCard } from "@/components/note-card";
import {
  useNotes,
  useNotesCreateMutation,
  useNotesDeleteMutation,
  useNotesEditMutation,
} from "@/hooks/notes";
import styles from "@/styles/list.module.css";

import type { Note, NoteFormData } from "@/types/note";

export function Notebook() {
  const { notes = [], isPending, isError } = useNotes();
  const createMutation = useNotesCreateMutation();
  const editMutation = useNotesEditMutation();
  const deleteMutation = useNotesDeleteMutation();

  const handleCreate = (formData: NoteFormData) => {
    createMutation.mutate(formData);
  };

  const handleEdit = (note: Note) => {
    editMutation.mutate(note);
  };

  const handleDelete = (id: number) => {
    deleteMutation.mutate(id);
  };

  if (isPending) {
    return <Loading />;
  }

  if (isError) {
    return <AppError />;
  }

  return (
    <div>
      <h1>Bloco de notas</h1>
      <div className={styles.listGrid}>
        {notes.map((note: Note) => (
          <NoteCard
            key={note.id}
            note={note}
            onDelete={handleDelete}
            onEdit={handleEdit}
          />
        ))}

        <CreateNoteCard onCreate={handleCreate} />
      </div>
    </div>
  );
}
