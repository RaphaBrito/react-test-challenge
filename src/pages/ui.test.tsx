import "@/test/utils";

import { QueryClient, QueryClientProvider } from "@tanstack/react-query";
import { render, waitFor, screen } from "@testing-library/react";
import { userEvent } from "@testing-library/user-event";
import jsonServer from "json-server";

import * as contacts from "@/hooks/contacts";
import * as notes from "@/hooks/notes";
import { Contacts } from "@/pages/contacts";
import { Notebook } from "@/pages/notebook";
import { ContactFormData } from "@/types/contact";
import { type NoteFormData } from "@/types/note";

const json = {
  contacts: [
    {
      id: 1,
      name: "Raphael",
      email: "brito@gmail.com",
      phone: "+558190836728",
    },
    {
      id: 2,
      name: "Cesar",
      email: "cesar@gmail.com",
      phone: "+558190836728",
    },
    {
      id: 3,
      name: "cesare",
      email: "Cesare@gmail.com",
      phone: "+558190836728",
    },
  ],
  notes: [
    {
      id: 1,
      title: "Lembrete",
      description: "Colocar presença na aula.",
    },
    {
      id: 2,
      title: "React",
      description: "Uma das bibliotecas mais utilizadas da atualidade.",
    },
    {
      id: 3,
      title: "CESAR",
      description: "Centro de Estudos e Sistemas Avançados do Recife.",
    },
  ],
};

const server = jsonServer.create();
export const router = jsonServer.router(json);
const middlewares = jsonServer.defaults();

server.use(middlewares);
server.use(router);
server.listen(5432, () => {
  console.log("JSON Server is running");
});

const resetDatabase = () => {
  router.db.setState(JSON.parse(JSON.stringify(json)) as typeof json);
  router.db.write();
};

function makeQueryClient(): QueryClient {
  return new QueryClient({
    defaultOptions: {
      queries: {
        retry: false,
      },
    },
  });
}

function ProviderWrapper({
  queryClient,
  children,
}: {
  queryClient: QueryClient;
  children: React.ReactNode;
}) {
  return (
    <QueryClientProvider client={queryClient}>{children}</QueryClientProvider>
  );
}

afterEach(() => {
  vi.restoreAllMocks();

  resetDatabase();
});

describe("ui tests", () => {
  it("notebook renders loading state", async () => {
    vi.spyOn(notes, "useNotes").mockReturnValue({
      notes: [],
      isPending: true,
      isError: false,
    });

    const queryClient = makeQueryClient();

    const { container } = render(
      <ProviderWrapper queryClient={queryClient}>
        <Notebook />
      </ProviderWrapper>,
    );

    await waitFor(() => {
      expect(queryClient.isFetching()).toBe(0);
    });

    await waitFor(() => {
      expect(screen.getByText("Carregando...")).toBeInTheDocument();
    });

    expect(container).toMatchSnapshot();
  });

  it("notebook renders error state", async () => {
    vi.spyOn(notes, "useNotes").mockReturnValue({
      notes: [],
      isPending: false,
      isError: true,
    });

    const queryClient = makeQueryClient();

    const { container } = render(
      <ProviderWrapper queryClient={queryClient}>
        <Notebook />
      </ProviderWrapper>,
    );

    await waitFor(() => {
      expect(queryClient.isFetching()).toBe(0);
    });

    await waitFor(() => {
      expect(
        screen.getByText("Que pena, algo de errado aconteceu :("),
      ).toBeInTheDocument();
    });

    expect(container).toMatchSnapshot();
  });

  it("notebook renders notes correctly", async () => {
    const queryClient = makeQueryClient();

    const { container } = render(
      <ProviderWrapper queryClient={queryClient}>
        <Notebook />
      </ProviderWrapper>,
    );

    await waitFor(() => {
      expect(queryClient.isFetching()).toBe(0);
    });

    await waitFor(() => {
      expect(screen.getByText(json.notes[0].title)).toBeInTheDocument();
    });

    expect(container).toMatchSnapshot();
  });

  it("notebook handles note creation", async () => {
    const formDataResponse = {
      title: "new note title",
      description: "new description",
    } satisfies NoteFormData;

    const queryClient = makeQueryClient();

    const { container } = render(
      <ProviderWrapper queryClient={queryClient}>
        <Notebook />
      </ProviderWrapper>,
    );

    await waitFor(() => {
      expect(queryClient.isFetching()).toBe(0);
    });

    await userEvent.click(screen.getByText("+"));

    const titleInput = screen.getByPlaceholderText("Title");
    expect(titleInput).toBeInTheDocument();

    await userEvent.type(titleInput, "new note title");
    expect(titleInput).toHaveValue("new note title");

    const descriptionInput = screen.getByPlaceholderText("Description");
    expect(descriptionInput).toBeInTheDocument();

    await userEvent.type(descriptionInput, "new description");
    expect(descriptionInput).toHaveValue("new description");

    await userEvent.click(screen.getByText("Salvar"));

    await waitFor(() => {
      expect(screen.getByText(formDataResponse.title)).toBeInTheDocument();
    });

    expect(container).toMatchSnapshot();
  });

  it("notebook handles note editing", async () => {
    const queryClient = makeQueryClient();

    const { container } = render(
      <ProviderWrapper queryClient={queryClient}>
        <Notebook />
      </ProviderWrapper>,
    );

    await waitFor(() => {
      expect(queryClient.isFetching()).toBe(0);
    });

    const firstEditButton = screen.getAllByText("Editar")[0];

    await userEvent.click(firstEditButton);

    const titleInput = screen.getByPlaceholderText("Title");
    expect(titleInput).toBeInTheDocument();

    await userEvent.clear(titleInput);
    await userEvent.type(titleInput, "edited note title");

    const descriptionInput = screen.getByPlaceholderText("Description");
    expect(descriptionInput).toBeInTheDocument();

    await userEvent.clear(descriptionInput);
    await userEvent.type(descriptionInput, "edited description");

    await userEvent.click(screen.getByText("Salvar"));

    await waitFor(() => {
      expect(screen.getByText("edited note title")).toBeInTheDocument();
    });

    expect(container).toMatchSnapshot();
  });

  it("notebook handles note deletion", async () => {
    const queryClient = makeQueryClient();

    const { container } = render(
      <ProviderWrapper queryClient={queryClient}>
        <Notebook />
      </ProviderWrapper>,
    );

    await waitFor(() => {
      expect(queryClient.isFetching()).toBe(0);
    });

    const firstTitle = screen.getByText("Lembrete");
    const firstDeleteButton = screen.getAllByText("Remover")[0];

    await userEvent.click(firstDeleteButton);

    await waitFor(() => {
      expect(firstTitle).not.toBeInTheDocument();
    });

    expect(container).toMatchSnapshot();
  });

  it("contacts renders loading state", async () => {
    vi.spyOn(contacts, "useContacts").mockReturnValue({
      contacts: [],
      isPending: true,
      isError: false,
    });

    const queryClient = makeQueryClient();

    const { container } = render(
      <ProviderWrapper queryClient={queryClient}>
        <Contacts />
      </ProviderWrapper>,
    );

    await waitFor(() => {
      expect(queryClient.isFetching()).toBe(0);
    });

    await waitFor(() => {
      expect(screen.getByText("Carregando...")).toBeInTheDocument();
    });

    expect(container).toMatchSnapshot();
  });

  it("renders contacts error state", async () => {
    vi.spyOn(contacts, "useContacts").mockReturnValue({
      contacts: [],
      isPending: false,
      isError: true,
    });

    const queryClient = makeQueryClient();

    const { container } = render(
      <ProviderWrapper queryClient={queryClient}>
        <Contacts />
      </ProviderWrapper>,
    );

    await waitFor(() => {
      expect(queryClient.isFetching()).toBe(0);
    });

    await waitFor(() => {
      expect(
        screen.getByText("Que pena, algo de errado aconteceu :("),
      ).toBeInTheDocument();
    });

    expect(container).toMatchSnapshot();
  });

  it("renders contacts correctly", async () => {
    const queryClient = makeQueryClient();

    const { container } = render(
      <ProviderWrapper queryClient={queryClient}>
        <Contacts />
      </ProviderWrapper>,
    );

    await waitFor(() => {
      expect(queryClient.isFetching()).toBe(0);
    });

    await waitFor(() => {
      expect(screen.getByText(json.contacts[0].name)).toBeInTheDocument();
    });

    expect(container).toMatchSnapshot();
  });

  it("handles contact creation", async () => {
    const formDataResponse = {
      name: "new name",
      email: "nem email",
      phone: "0800",
    } satisfies ContactFormData;

    const queryClient = makeQueryClient();

    const { container } = render(
      <ProviderWrapper queryClient={queryClient}>
        <Contacts />
      </ProviderWrapper>,
    );

    await waitFor(() => {
      expect(queryClient.isFetching()).toBe(0);
    });

    await userEvent.click(screen.getByText("+"));

    const nameInput = screen.getByPlaceholderText("Name");
    expect(nameInput).toBeInTheDocument();

    await userEvent.type(nameInput, "new name");
    expect(nameInput).toHaveValue("new name");

    const emailInput = screen.getByPlaceholderText("Email");
    expect(emailInput).toBeInTheDocument();

    await userEvent.type(emailInput, "new email");
    expect(emailInput).toHaveValue("new email");

    const phoneInput = screen.getByPlaceholderText("Phone");
    expect(phoneInput).toBeInTheDocument();

    await userEvent.type(phoneInput, "new phone");
    expect(phoneInput).toHaveValue("new phone");

    await userEvent.click(screen.getByText("Salvar"));

    await waitFor(() => {
      expect(screen.getByText(formDataResponse.name)).toBeInTheDocument();
    });

    expect(container).toMatchSnapshot();
  });

  it("handles contact editing", async () => {
    const queryClient = makeQueryClient();

    const { container } = render(
      <ProviderWrapper queryClient={queryClient}>
        <Contacts />
      </ProviderWrapper>,
    );

    await waitFor(() => {
      expect(queryClient.isFetching()).toBe(0);
    });

    const firstEditButton = screen.getAllByText("Editar")[0];

    await userEvent.click(firstEditButton);

    const nameInput = screen.getByPlaceholderText("Name");
    expect(nameInput).toBeInTheDocument();

    await userEvent.clear(nameInput);
    await userEvent.type(nameInput, "edited name");

    const emailInput = screen.getByPlaceholderText("Email");
    expect(emailInput).toBeInTheDocument();

    await userEvent.clear(emailInput);
    await userEvent.type(emailInput, "edited email");

    const phoneInput = screen.getByPlaceholderText("Phone");
    expect(phoneInput).toBeInTheDocument();

    await userEvent.clear(phoneInput);
    await userEvent.type(phoneInput, "edited phone");

    await userEvent.click(screen.getByText("Salvar"));

    await waitFor(() => {
      expect(screen.getByText("edited name")).toBeInTheDocument();
    });

    expect(container).toMatchSnapshot();
  });

  it("handles contact deletion", async () => {
    const queryClient = makeQueryClient();

    const { container } = render(
      <ProviderWrapper queryClient={queryClient}>
        <Contacts />
      </ProviderWrapper>,
    );

    await waitFor(() => {
      expect(queryClient.isFetching()).toBe(0);
    });

    const firstTitle = screen.getByText("Raphael");
    const firstDeleteButton = screen.getAllByText("Remover")[0];

    await userEvent.click(firstDeleteButton);

    await waitFor(() => {
      expect(firstTitle).not.toBeInTheDocument();
    });

    expect(container).toMatchSnapshot();
  });
});
