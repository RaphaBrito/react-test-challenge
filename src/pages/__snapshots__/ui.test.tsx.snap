// Vitest Snapshot v1, https://vitest.dev/guide/snapshot.html

exports[`ui tests > contacts handles note creation 1`] = `
<div>
  <div>
    <h1>
      Bloco de notas
    </h1>
    <div
      class="_listGrid_f34e7c"
    >
      <div
        class="_card_277467"
      >
        <h3>
          Lembrete
        </h3>
        <p>
          Colocar presença na aula.
        </p>
        <div
          class="_cardFooter_277467"
        >
          <button>
            Editar
          </button>
          <button>
            Remover
          </button>
        </div>
      </div>
      <div
        class="_card_277467"
      >
        <h3>
          React
        </h3>
        <p>
          Uma das bibliotecas mais utilizadas da atualidade.
        </p>
        <div
          class="_cardFooter_277467"
        >
          <button>
            Editar
          </button>
          <button>
            Remover
          </button>
        </div>
      </div>
      <div
        class="_card_277467"
      >
        <h3>
          CESAR
        </h3>
        <p>
          Centro de Estudos e Sistemas Avançados do Recife.
        </p>
        <div
          class="_cardFooter_277467"
        >
          <button>
            Editar
          </button>
          <button>
            Remover
          </button>
        </div>
      </div>
      <div
        class="_card_277467"
      >
        <h3>
          new note title
        </h3>
        <p>
          new description
        </p>
        <div
          class="_cardFooter_277467"
        >
          <button>
            Editar
          </button>
          <button>
            Remover
          </button>
        </div>
      </div>
      <button>
        +
      </button>
    </div>
  </div>
</div>
`;

exports[`ui tests > contacts handles note deletion 1`] = `
<div>
  <div>
    <h1>
      Bloco de notas
    </h1>
    <div
      class="_listGrid_f34e7c"
    >
      <div
        class="_card_277467"
      >
        <h3>
          React
        </h3>
        <p>
          Uma das bibliotecas mais utilizadas da atualidade.
        </p>
        <div
          class="_cardFooter_277467"
        >
          <button>
            Editar
          </button>
          <button>
            Remover
          </button>
        </div>
      </div>
      <div
        class="_card_277467"
      >
        <h3>
          CESAR
        </h3>
        <p>
          Centro de Estudos e Sistemas Avançados do Recife.
        </p>
        <div
          class="_cardFooter_277467"
        >
          <button>
            Editar
          </button>
          <button>
            Remover
          </button>
        </div>
      </div>
      <button>
        +
      </button>
    </div>
  </div>
</div>
`;

exports[`ui tests > contacts handles note editing 1`] = `
<div>
  <div>
    <h1>
      Bloco de notas
    </h1>
    <div
      class="_listGrid_f34e7c"
    >
      <div
        class="_card_277467"
      >
        <h3>
          edited note title
        </h3>
        <p>
          edited description
        </p>
        <div
          class="_cardFooter_277467"
        >
          <button>
            Editar
          </button>
          <button>
            Remover
          </button>
        </div>
      </div>
      <div
        class="_card_277467"
      >
        <h3>
          React
        </h3>
        <p>
          Uma das bibliotecas mais utilizadas da atualidade.
        </p>
        <div
          class="_cardFooter_277467"
        >
          <button>
            Editar
          </button>
          <button>
            Remover
          </button>
        </div>
      </div>
      <div
        class="_card_277467"
      >
        <h3>
          CESAR
        </h3>
        <p>
          Centro de Estudos e Sistemas Avançados do Recife.
        </p>
        <div
          class="_cardFooter_277467"
        >
          <button>
            Editar
          </button>
          <button>
            Remover
          </button>
        </div>
      </div>
      <button>
        +
      </button>
    </div>
  </div>
</div>
`;

exports[`ui tests > contacts renders error state 1`] = `
<div>
  <h1>
    Que pena, algo de errado aconteceu :(
  </h1>
</div>
`;

exports[`ui tests > contacts renders loading state 1`] = `
<div>
  <h1>
    Carregando...
  </h1>
</div>
`;

exports[`ui tests > contacts renders notes correctly 1`] = `
<div>
  <div>
    <h1>
      Bloco de notas
    </h1>
    <div
      class="_listGrid_f34e7c"
    >
      <div
        class="_card_277467"
      >
        <h3>
          Lembrete
        </h3>
        <p>
          Colocar presença na aula.
        </p>
        <div
          class="_cardFooter_277467"
        >
          <button>
            Editar
          </button>
          <button>
            Remover
          </button>
        </div>
      </div>
      <div
        class="_card_277467"
      >
        <h3>
          React
        </h3>
        <p>
          Uma das bibliotecas mais utilizadas da atualidade.
        </p>
        <div
          class="_cardFooter_277467"
        >
          <button>
            Editar
          </button>
          <button>
            Remover
          </button>
        </div>
      </div>
      <div
        class="_card_277467"
      >
        <h3>
          CESAR
        </h3>
        <p>
          Centro de Estudos e Sistemas Avançados do Recife.
        </p>
        <div
          class="_cardFooter_277467"
        >
          <button>
            Editar
          </button>
          <button>
            Remover
          </button>
        </div>
      </div>
      <button>
        +
      </button>
    </div>
  </div>
</div>
`;

exports[`ui tests > handles contact creation 1`] = `
<div>
  <div>
    <h1>
      Contatos
    </h1>
    <div
      class="_listGrid_f34e7c"
    >
      <div
        class="_card_277467"
      >
        <img
          alt="avatar"
          class="_cardImage_277467"
          src="/src/assets/images/avatar.png"
        />
        <h3>
          Raphael
        </h3>
        <div
          class="_cardContactInfo_277467"
        >
          <img
            alt="phone icon"
            height="20px"
            src="/src/assets/icons/phone.png"
          />
          <p>
            +558190836728
          </p>
        </div>
        <div
          class="_cardContactInfo_277467"
        >
          <img
            alt="e-mail icon"
            height="20px"
            src="/src/assets/icons/e-mail.png"
          />
          <p>
            brito@gmail.com
          </p>
        </div>
        <div
          class="_cardFooter_277467"
        >
          <button>
            Editar
          </button>
          <button>
            Remover
          </button>
        </div>
      </div>
      <div
        class="_card_277467"
      >
        <img
          alt="avatar"
          class="_cardImage_277467"
          src="/src/assets/images/avatar.png"
        />
        <h3>
          Cesar
        </h3>
        <div
          class="_cardContactInfo_277467"
        >
          <img
            alt="phone icon"
            height="20px"
            src="/src/assets/icons/phone.png"
          />
          <p>
            +558190836728
          </p>
        </div>
        <div
          class="_cardContactInfo_277467"
        >
          <img
            alt="e-mail icon"
            height="20px"
            src="/src/assets/icons/e-mail.png"
          />
          <p>
            cesar@gmail.com
          </p>
        </div>
        <div
          class="_cardFooter_277467"
        >
          <button>
            Editar
          </button>
          <button>
            Remover
          </button>
        </div>
      </div>
      <div
        class="_card_277467"
      >
        <img
          alt="avatar"
          class="_cardImage_277467"
          src="/src/assets/images/avatar.png"
        />
        <h3>
          cesare
        </h3>
        <div
          class="_cardContactInfo_277467"
        >
          <img
            alt="phone icon"
            height="20px"
            src="/src/assets/icons/phone.png"
          />
          <p>
            +558190836728
          </p>
        </div>
        <div
          class="_cardContactInfo_277467"
        >
          <img
            alt="e-mail icon"
            height="20px"
            src="/src/assets/icons/e-mail.png"
          />
          <p>
            Cesare@gmail.com
          </p>
        </div>
        <div
          class="_cardFooter_277467"
        >
          <button>
            Editar
          </button>
          <button>
            Remover
          </button>
        </div>
      </div>
      <div
        class="_card_277467"
      >
        <img
          alt="avatar"
          class="_cardImage_277467"
          src="/src/assets/images/avatar.png"
        />
        <h3>
          new name
        </h3>
        <div
          class="_cardContactInfo_277467"
        >
          <img
            alt="phone icon"
            height="20px"
            src="/src/assets/icons/phone.png"
          />
          <p>
            new phone
          </p>
        </div>
        <div
          class="_cardContactInfo_277467"
        >
          <img
            alt="e-mail icon"
            height="20px"
            src="/src/assets/icons/e-mail.png"
          />
          <p>
            new email
          </p>
        </div>
        <div
          class="_cardFooter_277467"
        >
          <button>
            Editar
          </button>
          <button>
            Remover
          </button>
        </div>
      </div>
      <button>
        +
      </button>
    </div>
  </div>
</div>
`;

exports[`ui tests > handles contact deletion 1`] = `
<div>
  <div>
    <h1>
      Contatos
    </h1>
    <div
      class="_listGrid_f34e7c"
    >
      <div
        class="_card_277467"
      >
        <img
          alt="avatar"
          class="_cardImage_277467"
          src="/src/assets/images/avatar.png"
        />
        <h3>
          Cesar
        </h3>
        <div
          class="_cardContactInfo_277467"
        >
          <img
            alt="phone icon"
            height="20px"
            src="/src/assets/icons/phone.png"
          />
          <p>
            +558190836728
          </p>
        </div>
        <div
          class="_cardContactInfo_277467"
        >
          <img
            alt="e-mail icon"
            height="20px"
            src="/src/assets/icons/e-mail.png"
          />
          <p>
            cesar@gmail.com
          </p>
        </div>
        <div
          class="_cardFooter_277467"
        >
          <button>
            Editar
          </button>
          <button>
            Remover
          </button>
        </div>
      </div>
      <div
        class="_card_277467"
      >
        <img
          alt="avatar"
          class="_cardImage_277467"
          src="/src/assets/images/avatar.png"
        />
        <h3>
          cesare
        </h3>
        <div
          class="_cardContactInfo_277467"
        >
          <img
            alt="phone icon"
            height="20px"
            src="/src/assets/icons/phone.png"
          />
          <p>
            +558190836728
          </p>
        </div>
        <div
          class="_cardContactInfo_277467"
        >
          <img
            alt="e-mail icon"
            height="20px"
            src="/src/assets/icons/e-mail.png"
          />
          <p>
            Cesare@gmail.com
          </p>
        </div>
        <div
          class="_cardFooter_277467"
        >
          <button>
            Editar
          </button>
          <button>
            Remover
          </button>
        </div>
      </div>
      <button>
        +
      </button>
    </div>
  </div>
</div>
`;

exports[`ui tests > handles contact editing 1`] = `
<div>
  <div>
    <h1>
      Contatos
    </h1>
    <div
      class="_listGrid_f34e7c"
    >
      <div
        class="_card_277467"
      >
        <img
          alt="avatar"
          class="_cardImage_277467"
          src="/src/assets/images/avatar.png"
        />
        <h3>
          edited name
        </h3>
        <div
          class="_cardContactInfo_277467"
        >
          <img
            alt="phone icon"
            height="20px"
            src="/src/assets/icons/phone.png"
          />
          <p>
            edited phone
          </p>
        </div>
        <div
          class="_cardContactInfo_277467"
        >
          <img
            alt="e-mail icon"
            height="20px"
            src="/src/assets/icons/e-mail.png"
          />
          <p>
            edited email
          </p>
        </div>
        <div
          class="_cardFooter_277467"
        >
          <button>
            Editar
          </button>
          <button>
            Remover
          </button>
        </div>
      </div>
      <div
        class="_card_277467"
      >
        <img
          alt="avatar"
          class="_cardImage_277467"
          src="/src/assets/images/avatar.png"
        />
        <h3>
          Cesar
        </h3>
        <div
          class="_cardContactInfo_277467"
        >
          <img
            alt="phone icon"
            height="20px"
            src="/src/assets/icons/phone.png"
          />
          <p>
            +558190836728
          </p>
        </div>
        <div
          class="_cardContactInfo_277467"
        >
          <img
            alt="e-mail icon"
            height="20px"
            src="/src/assets/icons/e-mail.png"
          />
          <p>
            cesar@gmail.com
          </p>
        </div>
        <div
          class="_cardFooter_277467"
        >
          <button>
            Editar
          </button>
          <button>
            Remover
          </button>
        </div>
      </div>
      <div
        class="_card_277467"
      >
        <img
          alt="avatar"
          class="_cardImage_277467"
          src="/src/assets/images/avatar.png"
        />
        <h3>
          cesare
        </h3>
        <div
          class="_cardContactInfo_277467"
        >
          <img
            alt="phone icon"
            height="20px"
            src="/src/assets/icons/phone.png"
          />
          <p>
            +558190836728
          </p>
        </div>
        <div
          class="_cardContactInfo_277467"
        >
          <img
            alt="e-mail icon"
            height="20px"
            src="/src/assets/icons/e-mail.png"
          />
          <p>
            Cesare@gmail.com
          </p>
        </div>
        <div
          class="_cardFooter_277467"
        >
          <button>
            Editar
          </button>
          <button>
            Remover
          </button>
        </div>
      </div>
      <button>
        +
      </button>
    </div>
  </div>
</div>
`;

exports[`ui tests > notebook handles note creation 1`] = `
<div>
  <div>
    <h1>
      Bloco de notas
    </h1>
    <div
      class="_listGrid_f34e7c"
    >
      <div
        class="_card_277467"
      >
        <h3>
          Lembrete
        </h3>
        <p>
          Colocar presença na aula.
        </p>
        <div
          class="_cardFooter_277467"
        >
          <button>
            Editar
          </button>
          <button>
            Remover
          </button>
        </div>
      </div>
      <div
        class="_card_277467"
      >
        <h3>
          React
        </h3>
        <p>
          Uma das bibliotecas mais utilizadas da atualidade.
        </p>
        <div
          class="_cardFooter_277467"
        >
          <button>
            Editar
          </button>
          <button>
            Remover
          </button>
        </div>
      </div>
      <div
        class="_card_277467"
      >
        <h3>
          CESAR
        </h3>
        <p>
          Centro de Estudos e Sistemas Avançados do Recife.
        </p>
        <div
          class="_cardFooter_277467"
        >
          <button>
            Editar
          </button>
          <button>
            Remover
          </button>
        </div>
      </div>
      <div
        class="_card_277467"
      >
        <h3>
          new note title
        </h3>
        <p>
          new description
        </p>
        <div
          class="_cardFooter_277467"
        >
          <button>
            Editar
          </button>
          <button>
            Remover
          </button>
        </div>
      </div>
      <button>
        +
      </button>
    </div>
  </div>
</div>
`;

exports[`ui tests > notebook handles note deletion 1`] = `
<div>
  <div>
    <h1>
      Bloco de notas
    </h1>
    <div
      class="_listGrid_f34e7c"
    >
      <div
        class="_card_277467"
      >
        <h3>
          React
        </h3>
        <p>
          Uma das bibliotecas mais utilizadas da atualidade.
        </p>
        <div
          class="_cardFooter_277467"
        >
          <button>
            Editar
          </button>
          <button>
            Remover
          </button>
        </div>
      </div>
      <div
        class="_card_277467"
      >
        <h3>
          CESAR
        </h3>
        <p>
          Centro de Estudos e Sistemas Avançados do Recife.
        </p>
        <div
          class="_cardFooter_277467"
        >
          <button>
            Editar
          </button>
          <button>
            Remover
          </button>
        </div>
      </div>
      <button>
        +
      </button>
    </div>
  </div>
</div>
`;

exports[`ui tests > notebook handles note editing 1`] = `
<div>
  <div>
    <h1>
      Bloco de notas
    </h1>
    <div
      class="_listGrid_f34e7c"
    >
      <div
        class="_card_277467"
      >
        <h3>
          edited note title
        </h3>
        <p>
          edited description
        </p>
        <div
          class="_cardFooter_277467"
        >
          <button>
            Editar
          </button>
          <button>
            Remover
          </button>
        </div>
      </div>
      <div
        class="_card_277467"
      >
        <h3>
          React
        </h3>
        <p>
          Uma das bibliotecas mais utilizadas da atualidade.
        </p>
        <div
          class="_cardFooter_277467"
        >
          <button>
            Editar
          </button>
          <button>
            Remover
          </button>
        </div>
      </div>
      <div
        class="_card_277467"
      >
        <h3>
          CESAR
        </h3>
        <p>
          Centro de Estudos e Sistemas Avançados do Recife.
        </p>
        <div
          class="_cardFooter_277467"
        >
          <button>
            Editar
          </button>
          <button>
            Remover
          </button>
        </div>
      </div>
      <button>
        +
      </button>
    </div>
  </div>
</div>
`;

exports[`ui tests > notebook renders error state 1`] = `
<div>
  <h1>
    Que pena, algo de errado aconteceu :(
  </h1>
</div>
`;

exports[`ui tests > notebook renders loading state 1`] = `
<div>
  <h1>
    Carregando...
  </h1>
</div>
`;

exports[`ui tests > notebook renders notes correctly 1`] = `
<div>
  <div>
    <h1>
      Bloco de notas
    </h1>
    <div
      class="_listGrid_f34e7c"
    >
      <div
        class="_card_277467"
      >
        <h3>
          Lembrete
        </h3>
        <p>
          Colocar presença na aula.
        </p>
        <div
          class="_cardFooter_277467"
        >
          <button>
            Editar
          </button>
          <button>
            Remover
          </button>
        </div>
      </div>
      <div
        class="_card_277467"
      >
        <h3>
          React
        </h3>
        <p>
          Uma das bibliotecas mais utilizadas da atualidade.
        </p>
        <div
          class="_cardFooter_277467"
        >
          <button>
            Editar
          </button>
          <button>
            Remover
          </button>
        </div>
      </div>
      <div
        class="_card_277467"
      >
        <h3>
          CESAR
        </h3>
        <p>
          Centro de Estudos e Sistemas Avançados do Recife.
        </p>
        <div
          class="_cardFooter_277467"
        >
          <button>
            Editar
          </button>
          <button>
            Remover
          </button>
        </div>
      </div>
      <button>
        +
      </button>
    </div>
  </div>
</div>
`;

exports[`ui tests > renders contacts correctly 1`] = `
<div>
  <div>
    <h1>
      Contatos
    </h1>
    <div
      class="_listGrid_f34e7c"
    >
      <div
        class="_card_277467"
      >
        <img
          alt="avatar"
          class="_cardImage_277467"
          src="/src/assets/images/avatar.png"
        />
        <h3>
          Raphael
        </h3>
        <div
          class="_cardContactInfo_277467"
        >
          <img
            alt="phone icon"
            height="20px"
            src="/src/assets/icons/phone.png"
          />
          <p>
            +558190836728
          </p>
        </div>
        <div
          class="_cardContactInfo_277467"
        >
          <img
            alt="e-mail icon"
            height="20px"
            src="/src/assets/icons/e-mail.png"
          />
          <p>
            brito@gmail.com
          </p>
        </div>
        <div
          class="_cardFooter_277467"
        >
          <button>
            Editar
          </button>
          <button>
            Remover
          </button>
        </div>
      </div>
      <div
        class="_card_277467"
      >
        <img
          alt="avatar"
          class="_cardImage_277467"
          src="/src/assets/images/avatar.png"
        />
        <h3>
          Cesar
        </h3>
        <div
          class="_cardContactInfo_277467"
        >
          <img
            alt="phone icon"
            height="20px"
            src="/src/assets/icons/phone.png"
          />
          <p>
            +558190836728
          </p>
        </div>
        <div
          class="_cardContactInfo_277467"
        >
          <img
            alt="e-mail icon"
            height="20px"
            src="/src/assets/icons/e-mail.png"
          />
          <p>
            cesar@gmail.com
          </p>
        </div>
        <div
          class="_cardFooter_277467"
        >
          <button>
            Editar
          </button>
          <button>
            Remover
          </button>
        </div>
      </div>
      <div
        class="_card_277467"
      >
        <img
          alt="avatar"
          class="_cardImage_277467"
          src="/src/assets/images/avatar.png"
        />
        <h3>
          cesare
        </h3>
        <div
          class="_cardContactInfo_277467"
        >
          <img
            alt="phone icon"
            height="20px"
            src="/src/assets/icons/phone.png"
          />
          <p>
            +558190836728
          </p>
        </div>
        <div
          class="_cardContactInfo_277467"
        >
          <img
            alt="e-mail icon"
            height="20px"
            src="/src/assets/icons/e-mail.png"
          />
          <p>
            Cesare@gmail.com
          </p>
        </div>
        <div
          class="_cardFooter_277467"
        >
          <button>
            Editar
          </button>
          <button>
            Remover
          </button>
        </div>
      </div>
      <button>
        +
      </button>
    </div>
  </div>
</div>
`;

exports[`ui tests > renders contacts error state 1`] = `
<div>
  <h1>
    Que pena, algo de errado aconteceu :(
  </h1>
</div>
`;
